### PR TITLE
Fix incompatible JRE version warning which is popped up while profiling a Ballerina program

### LINKED ISSUE #41548: [Bug]: Ballerina profiler gives an unnecessary warning there exist Java 20 installed in the developer machine
### Description

See the output:
```
================================================================================
Ballerina Profiler: Profiling...
================================================================================
Note: This is an experimental feature, which supports only a limited set of functionality.
[1/6] Initializing...
[2/6] Copying executable...
[3/6] Performing analysis...
[4/6] Instrumenting functions...
 ○ Instrumented module count: 46
 ○ Instrumented function count: 3079
[5/6] Running executable...
WARNING: Incompatible JRE version '20.0.1' found. This ballerina program supports running on JRE version '17.0.*'
```

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #41548

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ public class RunProfilerTask implements Task {
     private final PrintStream err;
     private static final String JAVA_OPTS = "JAVA_OPTS";
     private static final String CURRENT_DIR_KEY = "current.dir";
+    private static final String JAVA_COMMAND = "java.command";
```
Comment (warunalakshitha) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java:55]: There are other tasks like executable task, test task also use same system property. Shall move this to common util class for all tasks and refactor.

Comment (warunalakshitha) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java:55]: Or we can remove the constant and use the string directly as other tasks to be consistant.

**Review by HindujaB — APPROVED** [review]
LGTM

File: bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/codegen/ProfilerMethodWrapper.java
Comment (Nadeeshan96) [inline bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/codegen/ProfilerMethodWrapper.java:1]: Have we searched for other places also we just use string `"java"`? if there are more we need to fix those too.

Comment (gabilang) [reply bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/codegen/ProfilerMethodWrapper.java:1]: No other places found.

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**